app/controllers/users_controller.rb

```diff
@@ -0,0 +1,31 @@
+class UsersController < ApplicationController
+  skip_before_action :ensure_current_user
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    user = User.new(user_params)
+    if user.save
+      session[:user_id] = user.id
+      redirect_to root_path
+    else
+      render :new
+    end
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+    @questions = @user.questions
+    @answers = @user.answers
+    @comments = @user.comments
```

Comment:
Again, you can just use @user.questions etc in the view. You don't need these extra variables.